### ISSUE TITLE
Agent behind a Proxy

### ISSUE THREAD

**Issue by sergej-reiser** [MicrosoftDocs/vsts-docs#1145]
When using an Agent behind a corporate proxy, I have to explicitly set the proxy configuration inside the vsts gradle task definition, by adding parameter options like this:
`gradlew -Dhttp.proxyHost=proxy.mycompany.com -Dhttp.proxyPort=3128 myTask`

This of course leads to build failures, when switching to other non-proxy agents.

According to the [VSTS How-To-Guides](https://docs.microsoft.com/en-us/vsts/pipelines/agents/proxy?view=vsts&tabs=windows#how-the-agent-handles-the-proxy-within-a-build-or-release-job), every task is able to obtain proxy configs from the agent: 
> Agent will expose proxy configuration via environment variables for every task execution, task author need to use vsts-task-lib methods to retrieve back proxy configuration and handle proxy with their task.

Wouldn't it be possible to add the proxy params automatically?

**Comment by chrisrpatterson** [comment]
Yes it is possible for the task to do this it just requires an update to the task to read the proxy settings when they are set. Can you open an issue on the vsts-tasks repo for this.
